File: mobile/src/main/java/com/windscribe/mobile/ticket/SendTicketPresenter.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.ticket;

import android.content.Context;

import com.windscribe.vpn.api.response.QueryType;

public interface SendTicketPresenter {

    void init();

    void onInputChanged(String email, String subject, String message);

    void onQueryTypeSelected(QueryType queryType);

    void onSendTicketClicked(String email, String subject, String message);

    void setTheme(Context context);
}

File: mobile/src/main/java/com/windscribe/mobile/ticket/SendTicketPresenter.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.ticket

import android.content.Context
import com.windscribe.vpn.api.response.QueryType

interface SendTicketPresenter {
    fun init()
    fun onInputChanged(email: String, subject: String, message: String)
    fun onQueryTypeSelected(queryType: QueryType)
    fun onSendTicketClicked(email: String, subject: String, message: String)
    fun setTheme(context: Context)
}